src/core/workspace/WorkspaceResolver.ts

```diff
@@ -251,11 +253,77 @@ export class WorkspaceResolver {
 export const workspaceResolver = new WorkspaceResolver()
 
 /**
- * Phase 0: Convenience function for easy migration from path.resolve()
- * This is what we'll use to replace existing path.resolve(cwd, ...) calls
+ * Result type for multi-root workspace path resolution
  */
-export function resolveWorkspacePath(cwd: string, relativePath: string, context?: string): string {
-	return workspaceResolver.resolveWorkspacePath(cwd, relativePath, context) as string
+export interface WorkspacePathResult {
+	absolutePath: string
+	displayPath: string
+	resolvedPath: string
+}
+
+/**
+ * Configuration for workspace path resolution
+ */
+export interface WorkspaceConfig {
+	cwd: string
+	isMultiRootEnabled?: boolean
+	workspaceManager?: any
+}
+
+/**
+ * Enhanced workspace path resolution that handles both single and multi-root workspaces
+ *
+ * @param cwdOrConfig - Either a string (for backward compatibility) or a config object
+ * @param relativePath - The relative path to resolve
+ * @param context - Component/handler name for tracking usage
+ * @returns Either a string (for backward compatibility) or a WorkspacePathResult object
+ */
+export function resolveWorkspacePath(
+	cwdOrConfig: string | WorkspaceConfig,
+	relativePath: string,
+	context?: string,
+): string | WorkspacePathResult {
```

Comment:
Why don't we just always return WorkspacePathResult and we just get the absolute path from there?
This makes it type safe as we don't have to use "as string" and is clear what it's being returned. wdyt?  

```suggestion
): WorkspacePathResult {
```